adapter/src/main/kotlin/de/fraunhofer/iem/kpiCalculator/adapter/KpiAdapter.kt

```diff
@@ -9,32 +9,17 @@
 
 package de.fraunhofer.iem.kpiCalculator.adapter
 
+import de.fraunhofer.iem.kpiCalculator.adapter.tools.tlc.TechLagResult
 import de.fraunhofer.iem.kpiCalculator.model.kpi.RawValueKpi
 
-/***
- * Adapter that transforms a list of elements into a list of raw KPI values.
- * @param T The type of the element that shall be transformed.
- */
-interface KpiAdapter<T> {
-    /**
-     * Transforms a specified list of elements into a list of raw KPI values.
-     * In the case of failure, the result may contain elements indicating the status of the failure.
-     * @param data The elements that shall be transformed
-     * @return The transformed elements.
-     */
-    fun transformDataToKpi(data: Collection<T>): Collection<AdapterResult>
-    /**
-     * Transforms an elements into a list of raw KPI values.
-     * In the case of failure, the result indicates the status of the failures.
-     * @param data The element that shall be transformed
-     * @return The transformed elements.
-     */
-    fun transformDataToKpi(data: T): Collection<AdapterResult> = transformDataToKpi(listOf(data))
-}
 
 enum class ErrorType { DATA_VALIDATION_ERROR }
 
 sealed class AdapterResult {
-    data class Success(val rawValueKpi: RawValueKpi) : AdapterResult()
+    sealed class Success(val rawValueKpi: RawValueKpi) : AdapterResult() {
+        class Kpi(rawValueKpi: RawValueKpi) : Success(rawValueKpi)
```

Comment:
do you want these to be data classes?

Reply:
yes but it was complicated.. :D 
the following code is not allowed: 
`        data class Kpi(override val rawValueKpi: RawValueKpi) : Success(rawValueKpi)`
as we may not have override modifiers in primary constructors. We also must only have val / var in primary constructors. So we can't do the pattern we do with a class `class Kpi(rawValueKpi: RawValueKpi) : Success(rawValueKpi)` and "hand down" the rawValueKpi to its superclass.